test/modules/TestResultCoordinator/Reports/EdgeHubRestartTest/EdgeHubRestartDirectMethodReportGenerator.cs

```diff
@@ -0,0 +1,258 @@
+
+                // Check if the current dm is passing
+                bool isCurrentDirectMethodPassing = true;
+                EdgeHubRestartDirectMethodResult senderResult = JsonConvert.DeserializeObject<EdgeHubRestartDirectMethodResult>(this.SenderTestResults.Current.Result);
+
+                // Verified the sequence numbers are the same
+                isCurrentDirectMethodPassing &= senderSeqNum == receiverSeqNum;
+
+                // Verified the sequence numbers are incremental
+                isCurrentDirectMethodPassing &= senderSeqNum > previousSeqNum;
+
+                // Make sure the status code is passing
+                isCurrentDirectMethodPassing &= senderResult.DirectMethodCompletedStatusCode == HttpStatusCode.OK;
+
+                // If the current DM result is passed, increment the count for a good dm
+                passedDirectMethodCount += isCurrentDirectMethodPassing ? 1UL : 0UL;
+                previousSeqNum = senderSeqNum;
+
+                (senderResultCount, hasSenderResult, senderSeqNum) = await this.MoveNextSenderResultAsync(senderResultCount);
+                (receiverResultCount, hasReceiverResult, receiverSeqNum) = await this.MoveNextReceiverResultAsync(receiverResultCount);
+            }
+
+            (senderResultCount, _, _) = await this.IterateResultToSequenceNumberAsync(
+                hasSenderResult,
+                this.MoveNextSenderResultAsync,
+                long.MaxValue,
+                senderResultCount);
+
+            (receiverResultCount, _, _) = await this.IterateResultToSequenceNumberAsync(
+                hasReceiverResult,
+                this.MoveNextReceiverResultAsync,
+                long.MaxValue,
+                receiverResultCount);
+
+            EdgeHubRestartStatistics edgeHubRestartStatistics = new EdgeHubRestartStatistics(this.completedStatusHistogram);
+            edgeHubRestartStatistics.CalculateStatistic();
+            Logger.LogInformation(JsonConvert.SerializeObject(edgeHubRestartStatistics));
+
+            return new EdgeHubRestartDirectMethodReport(
+                this.TrackingId,
+                this.TestReportType.ToString(),
+                passedDirectMethodCount,
+                this.SenderSource,
+                this.ReceiverSource,
+                senderResultCount,
+                receiverResultCount,
+                edgeHubRestartStatistics.MedianPeriod);
+        }
+
+        async Task<(ulong resultCount, bool hasValue, long sequenceNum)> IterateResultToSequenceNumberAsync(
+            bool hasValue,
+            MoveNextResultAsync MoveNextResultAsync,
+            long targetSequenceNumber,
+            ulong resultCount)
+        {
+            long seqNum = 0;
+
+            while (hasValue && (seqNum < targetSequenceNumber))
+            {
+                (resultCount, hasValue, seqNum) = await MoveNextResultAsync(resultCount);
+            }
+
+            return (resultCount: resultCount,
```

Comment:
should return original sequence number rather or return null or option type, but not zero.

Reply:
Given the current use cases for the sequence number, making sequence number nullable/Option<> does not add anything to the code since the `hasValue` is required to be checked before accessing the seqNum to get a meaningful value.